screens/Menu/ShuttleBus.js

```diff
@@ -1,32 +1,245 @@
+                Hours: diffHours + "h",
+                Minutes: "0" + Math.abs(diffMinutes) + "min"
+            });
+        }
+        else if (diffHours < 1) {
+            return ({
+                Hours: "",
+                Minutes: Math.abs(diffMinutes) + " min"
+            });
+        }
+        else if (Math.abs(diffMinutes) < 1) {
+            return ({
+                Hours: diffHours + " h",
+                Minutes: ""
+            });
+        }
+        else {
+            return ({
+                Hours: diffHours + "h",
+                Minutes: Math.abs(diffMinutes) + "min"
+            });
+        }
+
+    };
+
+    /**
+     * Function that returns an object containing the id, timeDifference as well as the time of the next buses on schedule
+     * @param {*} list 
+     */
+    const fetchNextStops = (list) => {
+        var nextStops = [];
+        var results = [];
+        let index = 0;
+        for (var key in list) {
+            if (list[key].hour >= currentTime.Hour && list[key].minutes >= currentTime.Minutes) {
+                nextStops = list.slice(index, list.length - 1);
+                break;
+            }
+            index++;
+        }
+        if (nextStops.length > 0) {
+            results = nextStops.map((element, index) => {
+                return ({
+                    id: index.toString(),
+                    timeDifference: calculateTimeDifference(element),
+                    hour: element.hour,
+                    minutes: element.minutes
+                });
+            });
+        }
+        return results;
+    };
+
+    /**
+     * Function that sorts through the shuttle bus schedule and compares the time/day to the current time/day
+     */
+    const getNextStops = () => {
+        var scheduleTimesSGW = [];
+        var scheduleTimesLoyola = [];
+        const sgwCampus = "SGW";
+        const loyolaCampus = "Loyola";
+        if (currentDay && weekDays.includes(currentDay.toString()) && !friday.includes(currentDay.toString()) && !weekends.includes(currentDay.toString())) {
+            scheduleTimesSGW = getShuttleBusTimes[sgwCampus].MondayToThursday;
+            scheduleTimesLoyola = getShuttleBusTimes[loyolaCampus].MondayToThursday;
+        }
+        else if (currentDay && friday.includes(currentDay.toString())) {
+            scheduleTimesSGW = getShuttleBusTimes[sgwCampus].Friday;
+            scheduleTimesLoyola = getShuttleBusTimes[loyolaCampus].Friday;
+        }
+        else {
+            return alert("There is no shuttle bus on weekends. Please check back during the week!");
+        }
+        var nextStopsSGW = fetchNextStops(scheduleTimesSGW);
+        var nextStopsLoyola = fetchNextStops(scheduleTimesLoyola);
+        return ({
+            sgwStops: nextStopsSGW,
+            LoyolaStops: nextStopsLoyola
+        });
+    };
+
+    useEffect(() => {
+        const intervalId = setInterval(() => {
+            getCurrentTime();
+        }, 100);
```

Comment:
Can't we change this to every 60,000ms instead?

Reply:
Tried it and it does not work. It doesn't fetch the data the first time

Reply:
You don't need currentDay (if you follow the logic I suggested earlier) and campus (we fetch both campuses in the beginning) anymore so your useEffect needs to change like the following: Change your useEffect to this one instead. 
You will need to Modify the getNextStop() method accordingly (to avoid crashes). 

```
    useEffect(() => {
        const intervalId = setInterval(() => {
            getCurrentTime();
        }, 100);
        if (currentTime) {
            setResults(getNextStops());
        }
        return () => clearInterval(intervalId);
    }, [currentTime]);
```